test/TestCaseReader.cpp

```diff
@@ -137,6 +146,56 @@ pair<SourceMap, size_t> TestCaseReader::parseSourcesAndSettingsWithLineNumber(is
 				if (sources.count(currentSourceName))
 					BOOST_THROW_EXCEPTION(runtime_error("Multiple definitions of test source \"" + currentSourceName + "\"."));
 			}
+			else if (boost::algorithm::starts_with(line, externalSourceDelimiterStart) && boost::algorithm::ends_with(line, sourceDelimiterEnd))
+			{
+				string externalSource = boost::trim_copy(line.substr(
+					externalSourceDelimiterStart.size(),
+					line.size() - sourceDelimiterEnd.size() - externalSourceDelimiterStart.size()
+				));
+				string externalSourceName = externalSource;
+
+				// Does the external source define a remapping?
+				size_t remappingPos = externalSource.find('=');
+				if (remappingPos != string::npos)
+				{
+					externalSourceName = boost::trim_copy(externalSource.substr(0, remappingPos));
+					externalSource = boost::trim_copy(externalSource.substr(remappingPos + 1));
+				}
+
+				fs::path testCaseParentDir(m_fileName);
+				testCaseParentDir = canonical(testCaseParentDir).parent_path();
+				fs::path externalSourcePath(externalSource);
+				if (!fs::path(externalSource).is_relative())
+					BOOST_THROW_EXCEPTION(runtime_error(string("External Source need to be relative.")));
+				fs::path externalSourceFullPath = testCaseParentDir / externalSourcePath;
+
+				string externalSourceContent;
+				if (fs::exists(externalSourceFullPath))
+					externalSourceContent = util::readFileAsString(externalSourceFullPath.string());
+				else
+					BOOST_THROW_EXCEPTION(runtime_error("External Source '" + externalSourcePath.string() + "' not found."));
+
+				if (!externalSourceName.empty())
+				{
+					ErrorList errorList;
+					ErrorReporter errorReporter(errorList);
+					shared_ptr<Scanner> scanner = make_shared<Scanner>(CharStream(externalSourceContent, externalSourceName));
+					ASTPointer<SourceUnit> sourceUnit = Parser(errorReporter, evmVersion).parse(scanner);
+					solAssert(sourceUnit != nullptr, "");
+					if (sourceUnit)
+						for (auto const* import: ASTNode::filteredNodes<ImportDirective>(sourceUnit->nodes()))
+						{
+							fs::path externalSourceParentDir(externalSource);
+							externalSourceParentDir = externalSourceParentDir.parent_path();
+							solAssert(externalSourceParentDir.is_relative() && fs::path(import->path()).is_relative(), "");
+							string importedSourceContent = util::readFileAsString((testCaseParentDir / externalSourceParentDir / import->path()).string());
```

Comment:
This line is too long
```suggestion
							string importedSourceContent = util::readFileAsString(
								(testCaseParentDir / externalSourceParentDir / import->path()).string()
							);
```